src/datasim_ui/handlers.cljs

```diff
@@ -84,6 +86,44 @@
    (update-in db [::db/options :options/visible]
               not)))
 
+(re-frame/reg-event-db
+  :validation/show
+  global-interceptors
+  (fn [db _]
+    (assoc-in db [::db/validation :validation/visible]
+              true)))
+
+(re-frame/reg-event-db
+ :validation/hide
+ global-interceptors
+ (fn [db _]
+   (assoc-in db [::db/validation :validation/visible]
+             false)))
+
+(re-frame/reg-event-db
+ :validation/data
+ global-interceptors
+ (fn [db [_ data]]
+   (assoc-in db [::db/validation :validation/data]
+             (map (fn [error]
+                    (merge {:visible false
+                            :id (random-uuid)}
+                           error))
```

Comment:
Just for clarity - this is adding a default id and visibility to error entries, which are not in the server return. If they exist the existing ones override.

Reply:
binding the input as `[_ data]` is a little confusing if it's really gonna be `[_ errors]`

Reply:
done